internal/adapters/dataproviders/database/mongo/pegin.go

```diff
@@ -208,3 +209,30 @@ func (repo *peginMongoRepository) DeleteQuotes(ctx context.Context, quotes []str
 	}
 	return uint(peginResult.DeletedCount + retainedResult.DeletedCount + creationDataResult.DeletedCount), nil
 }
+
+func (repo *peginMongoRepository) GetRetainedQuotesForAddress(ctx context.Context, address string) ([]quote.RetainedPeginQuote, error) {
+	result := make([]quote.RetainedPeginQuote, 0)
+	dbCtx, cancel := context.WithTimeout(ctx, repo.conn.timeout)
+	defer cancel()
+
+	collection := repo.conn.Collection(RetainedPeginQuoteCollection)
+	filter := bson.D{
+		primitive.E{Key: "owner_account_address", Value: address},
+		primitive.E{Key: "state", Value: bson.D{
+			primitive.E{Key: "$in", Value: []quote.PeginState{
+				quote.PeginStateWaitingForDeposit,
```

Comment:
This shouldn't be hardcoded here because then the consumer of the function needs to guess that it only fetches with this two states, it is preferable to receive them through a variadic argument

Reply:
I modified the function to receive the variadic argument.